pkg/controller/druid/handler.go

```diff
@@ -151,6 +154,18 @@ func deployDruidCluster(sdk client.Client, m *v1alpha1.Druid) error {
 				return err
 			}
 		}
+
+		// Create HPA Spec
+		if nodeSpec.AutoScaler != nil {
```

Comment:
can we create this before  creating the StatefulSet to potentially StatefulSet to know about  it as soon as that gets created ?

Reply:
hpa needs to know about the statefulset, the ``` scaleTargetRef``` specifies which target to look to.  So what i feel ideally is a target gets created ( in our case its statefullset ) and then hpa gets attached to that target.